`ob run` does not work with own Prelude

#### To Reproduce

1. ob init
2. Create `common/src/Prelude.hs` add it to `common.cabal`
3. run `ob run`.

#### Error

```
Loading ghci -ignore-dot-ghci -no-user-package-db -package-env - -F -pgmF /nix/store/gd99xxp41i5dllhpx341rxscgbyzlpqn-obelisk-command-0.8.0.0/bin/.ob-wrapped -optF __preprocessor-apply-packages -optF backend/backend.cabal -optF common/common.cabal -optF frontend/frontend.cabal -ibackend/src:common/src:frontend/src -ghci-script /tmp/ob-ghci-f5ba8bc8df8585b5/.ghci ...
GHCi, version 8.6.5: http://www.haskell.org/ghc/  :? for help

<interactive>:1:1: error:
    attempting to use module ‘main:Prelude’ (common/src/Prelude.hs) which is not loaded
Command "ghci -ignore-dot-ghci -no-user-package-db -package-env - -F -pgmF /nix/store/gd99xxp41i5dllhpx341rxscgbyzlpqn-obelisk-command-0.8.0.0/bin/.ob-wrapped -optF __preprocessor-apply-packages -optF backend/backend.cabal -optF common/common.cabal -optF frontend/frontend.cabal -ibackend/src:common/src:frontend/src -ghci-script /tmp/ob-ghci-f5ba8bc8df8585b5/.ghci" exited unexpectedly with error message:     attempting to use module ‘main:Prelude’ (common/src/Prelude.hs) which is not loaded
```

This has something to do with a special handling of the name `Prelude` and is a regression. I didn‘t have the problem before bumping to master today.

This also affects `ob profile`, pasting it here because the error looks different:
```
building '/nix/store/2pb5m75z2flsfqsxxf77llcbylra5bmp-ob-run.drv'...
[1 of 1] Compiling Main             ( ob-run.hs, ob-run.o )

ob-run.hs:1:8: error:
    Ambiguous module name `Prelude':
      it was found in multiple packages: base-4.12.0.0 common-0.1
  |
1 | module Main where
  |        ^^^^
```

--- a/lib/command/src/Obelisk/Command/Run.hs
+++ b/lib/command/src/Obelisk/Command/Run.hs
@@ -31,7 +31,7 @@ import qualified Data.List.NonEmpty as NE
 import Data.Map (Map)
 import qualified Data.Map as Map
 import qualified Data.Map.Monoidal as MMap
-import Data.Maybe (catMaybes, fromMaybe, mapMaybe, maybeToList)
+import Data.Maybe (catMaybes, fromMaybe, mapMaybe)
 import Data.Ord (comparing)
 import Data.Set (Set)
 import qualified Data.Set as Set
@@ -163,12 +163,12 @@ profile profileBasePattern rtsFlags = withProjectRoot "." $ \root -> do
 run :: MonadObelisk m => FilePath -> PathTree Interpret -> m ()
 run root interpretPaths = do
   pkgs <- getParsedLocalPkgs root interpretPaths
-  withGhciScript pkgs $ \dotGhciPath -> do
-    assets <- findProjectAssets root
-    putLog Debug $ "Assets impurely loaded from: " <> assets
-    ghciArgs <- getGhciSessionSettings pkgs root True
-    freePort <- getFreePort
-    runGhcid root True (ghciArgs <> mkGhciScriptArg dotGhciPath) pkgs $ Just $ unwords
+  assets <- findProjectAssets root
+  putLog Debug $ "Assets impurely loaded from: " <> assets
+  ghciArgs <- getGhciSessionSettings pkgs root True
+  freePort <- getFreePort
+  withGhciScriptArgs pkgs $ \dotGhciArgs -> do
+    runGhcid root True (ghciArgs <> dotGhciArgs) pkgs $ Just $ unwords
       [ "Obelisk.Run.run"
       , show freePort
       , "(Obelisk.Run.runServeAsset " ++ show assets ++ ")"
@@ -180,15 +180,15 @@ runRepl :: MonadObelisk m => FilePath -> PathTree Interpret -> m ()
 runRepl root interpretPaths = do
   pkgs <- getParsedLocalPkgs root interpretPaths
   ghciArgs <- getGhciSessionSettings pkgs "." True
-  withGhciScript pkgs $ \dotGhciPath ->
-    runGhciRepl root pkgs (ghciArgs <> mkGhciScriptArg dotGhciPath)
+  withGhciScriptArgs pkgs $ \dotGhciArgs ->
+    runGhciRepl root pkgs (ghciArgs <> dotGhciArgs)
 
 runWatch :: MonadObelisk m => FilePath -> PathTree Interpret -> m ()
 runWatch root interpretPaths = do
   pkgs <- getParsedLocalPkgs root interpretPaths
   ghciArgs <- getGhciSessionSettings pkgs root True
-  withGhciScript pkgs $ \dotGhciPath ->
-    runGhcid root True (ghciArgs <> mkGhciScriptArg dotGhciPath) pkgs Nothing
+  withGhciScriptArgs pkgs $ \dotGhciArgs ->
+    runGhcid root True (ghciArgs <> dotGhciArgs) pkgs Nothing
 
 exportGhciConfig :: MonadObelisk m => Bool -> FilePath -> PathTree Interpret -> m [String]
 exportGhciConfig useRelativePaths root interpretPaths = do
@@ -395,13 +395,29 @@ parsePackagesOrFail dirs' = do
 packageInfoToNamePathMap :: Foldable f => f CabalPackageInfo -> Map Text FilePath
 packageInfoToNamePathMap = Map.fromList . map (_cabalPackageInfo_packageName &&& _cabalPackageInfo_packageRoot) . toList
 
+-- Like 'withGhciScript' but provides the precise ghci arguments to add to a ghci session
+withGhciScriptArgs
+  :: (MonadObelisk m, Foldable f)
+  => f CabalPackageInfo -- ^ List of packages to load into ghci
+  -> ([String] -> m ()) -- ^ Action to run with the extra ghci arguments
+  -> m ()
+withGhciScriptArgs packageInfos f = withGhciScript loadPreludeManually packageInfos $ \fp -> f ["-XNoImplicitPrelude", "-ghci-script", fp]
+  where
+    -- These lines must be first and allow the session to support a custom Prelude when @-XNoImplicitPrelude@
+    -- is passed to the ghci session.
+    loadPreludeManually =
+      [ ":add Prelude" -- @:add@ is used because it's less noisy when there is no custom Prelude
+      , ":set -XImplicitPrelude" -- Turn the default setting on
+      ]
+
 -- | Create ghci configuration to load the given packages
 withGhciScript
   :: (MonadObelisk m, Foldable f)
-  => f CabalPackageInfo -- ^ List of packages to load into ghci
+  => [String] -- ^ Commands to prefix to file
+  -> f CabalPackageInfo -- ^ List of packages to load into ghci
   -> (FilePath -> m ()) -- ^ Action to run with the path to generated temporary .ghci
   -> m ()
-withGhciScript (toList -> packageInfos) f =
+withGhciScript preCommands (toList -> packageInfos) f =
   withSystemTempDirectory "ob-ghci" $ \fp -> do
     let dotGhciPath = fp </> ".ghci"
     liftIO $ writeFile dotGhciPath dotGhci
@@ -413,7 +429,8 @@ withGhciScript (toList -> packageInfos) f =
       , [ "Backend" | "backend" `Set.member` packageNames ]
       , [ "Frontend" | "frontend" `Set.member` packageNames ]
       ]
-    dotGhci = unlines
+    dotGhci = unlines $
+      preCommands <>
       [ if null modulesToLoad then "" else ":load " <> unwords modulesToLoad
       , "import qualified Obelisk.Run"
       , "import qualified Frontend"
@@ -492,9 +509,6 @@ runGhcid root chdirToRoot ghciArgs (toList -> packages) mcmd =
     adjustRoot x = if chdirToRoot then makeRelative root x else x
     reloadFiles = map adjustRoot [root </> "config"]
     restartFiles = map (adjustRoot . _cabalPackageInfo_packageFile) packages
-
-mkGhciScriptArg :: FilePath -> [String]
-mkGhciScriptArg dotGhci = ["-ghci-script", dotGhci]
 
 getFreePort :: MonadIO m => m Socket.PortNumber
 getFreePort = liftIO $ Socket.withSocketsDo $ do

